slide-deck.js

```diff
@@ -230,6 +232,26 @@ class slideDeck extends HTMLElement {
     slideDeck.#adoptShadowStyles(this);
     this.#setDeckID();
 
+    // custom events
+    this.addEventListener('key-control', (e) => this.toggleKeyControl());
+    this.addEventListener('follow-active', (e) => this.toggleFollowActive());
+    this.addEventListener('full-screen', (e) => this.toggleFullScreen());
+
+    this.addEventListener('join', (e) => this.join());
+    this.addEventListener('start', (e) => this.start());
+    this.addEventListener('resume', (e) => this.resume());
+    this.addEventListener('reset', (e) => this.reset());
+    this.addEventListener('blank-slide', (e) => this.blankSlide());
+    this.addEventListener('join-as-speaker', (e) => this.joinAsSpeaker());
+
+    this.addEventListener('next', (e) => this.next());
+    this.addEventListener('previous', (e) => this.previous());
+    this.addEventListener('to-slide', (e) => this.toSlide(e.detail));
+    this.addEventListener('to-saved', (e) => this.toSavedSlide());
+    this.addEventListener('scroll-to-active', (e) => this.scrollToActive());
+  };
+
+  connectedCallback() {
```

Comment:
@jamesnw for listeners on the element itself, does it matter? That's the distinction that I thought I understood for when to use which. 

If we do put those in the connectedCallback, do we also need to remove them in disconnectedCallback?

Reply:
Agh yes, you're right, that is the distinction, and actually since `connectedCallback` can be called multiple times, leaving it in `constructor` means we can be confident the events are only added once.

If we did move it to `connectedCallback`, we would need to mirror the `add`s with `remove`s. Otherwise, if you remove an element from the dom but don't destroy it, and then re-add it to the dom, you would end up with duplicated event listeners.